From 5ac76dfd31d5e0d243b2fea12103b823acb22211 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Sat, 30 Sep 2023 17:30:37 +0200
Subject: [PATCH 494/954] feat(stm32mp21-fdts): add pinctrl DT file

File contains only SDMMC1 pins for the moment and will be filled with
peripheral pins nodes.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 53d4b443d5d56faa6553c0598d9fd0d380cd74fb feat(stm32mp21-fdts): add pinctrl DT file]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: Ie4b18afa7ad7872ef6b4570623fa185c177f1af1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/356315
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 fdts/stm32mp21-pinctrl.dtsi | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)
 create mode 100644 fdts/stm32mp21-pinctrl.dtsi

diff --git a/fdts/stm32mp21-pinctrl.dtsi b/fdts/stm32mp21-pinctrl.dtsi
new file mode 100644
index 0000000000..2fc2cd79e0
--- /dev/null
+++ b/fdts/stm32mp21-pinctrl.dtsi
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) 2024, STMicroelectronics - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+#include <dt-bindings/pinctrl/stm32-pinfunc.h>
+
+&pinctrl {
+	/omit-if-no-ref/
+	sdmmc1_b4_pins_a: sdmmc1-b4-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('E', 4, AF10)>, /* SDMMC1_D0 */
+				 <STM32_PINMUX('E', 5, AF10)>, /* SDMMC1_D1 */
+				 <STM32_PINMUX('E', 0, AF10)>, /* SDMMC1_D2 */
+				 <STM32_PINMUX('E', 1, AF10)>, /* SDMMC1_D3 */
+				 <STM32_PINMUX('E', 2, AF10)>; /* SDMMC1_CMD */
+			slew-rate = <2>;
+			drive-push-pull;
+			bias-disable;
+		};
+		pins2 {
+			pinmux = <STM32_PINMUX('E', 3, AF10)>; /* SDMMC1_CK */
+			slew-rate = <3>;
+			drive-push-pull;
+			bias-disable;
+		};
+	};
+};
-- 
2.53.0

